### PR TITLE
Test plutus-tx-tests on all supported GHC versions

### PR DESCRIPTION
**Pull request by zliu41**
Currently 9.2 and 9.6.

### CODE REVIEW

**Review by michaelpj — APPROVED** [review]
Great. Do we want to do the tests in `plutus-benchmark` also? They're also often good tests of the compiler.

File: plutus-tx-plugin/src/PlutusTx/Compiler/Names.hs [hunk 1]
```diff
@@ -49,34 +47,34 @@ on how it is printed.
 getUntidiedOccString :: GHC.Name -> String
 getUntidiedOccString n = dropWhileEnd isDigit (GHC.getOccString n)
 
-compileNameFresh :: MonadQuote m => GHC.Name -> m PLC.Name
+compileNameFresh :: (MonadQuote m) => GHC.Name -> m PLC.Name
```
[the remark this answers is not in the document]
Comment (michaelpj) [reply plutus-tx-plugin/src/PlutusTx/Compiler/Names.hs:50]: I assume you will revert before merging!

[the remark this answers is not in the document]
Comment (zliu41) [reply plutus-tx-plugin/src/PlutusTx/Compiler/Names.hs:50]: Nah, the formatting is good to do anyway.

**Review by zliu41 — COMMENTED** [review]
> Great. Do we want to do the tests in plutus-benchmark also? They're also often good tests of the compiler.

Yes, but after https://input-output.atlassian.net/browse/PLT-6722.

**Pull request merged**